lib/string.rb

```diff
@@ -2,4 +2,45 @@ class String
   def humanized_all_first_capitals
     self.humanize.split(' ').map{|w| w.capitalize}.join(' ')
   end
+  
+  def downcase_words
+    self.scan(/\b\w+\b/).map(&:downcase)
+  end
+  
+  # Methods below created to customly setup slugs from friendly_id gem
+  
+  def short_name
```

Comment:
just a thought for the future - but I wonder if ultimately what we are seeing here is the emergence of a new domain object, a Name class perhaps ...?